src/ert/_c_wrappers/job_queue/ext_job.py

```diff
@@ -1,263 +1,211 @@
+        parser.add("ENV").set_argc_minmax(1, 2)
+        parser.add("EXEC_ENV").set_argc_minmax(1, 2)
+        parser.add("DEFAULT").set_argc_minmax(2, 2)
+        parser.add("ARGLIST").set_argc_minmax(1, -1)
+        schema = parser.add("ARG_TYPE")
+        schema.set_argc_minmax(2, 2)
+        schema.iset_type(0, ContentTypeEnum.CONFIG_INT)
+
+        return parser.parse(
+            config_file,
+        )
+
+    @classmethod
+    def _read_str_keywords(cls, content_dict, config_content):
+        def set_nullable(keyword, key):
+            value = config_content.getValue(keyword)
+            if value == "null":
+                value = None
+            content_dict[key] = value
+
+        for key in cls._str_keywords:
+            if config_content.hasKey(key):
+                if key in ("STDIN", "STDOUT", "STDERR"):
+                    set_nullable(key, key.lower() + "_file")
+                else:
+                    set_nullable(key, key.lower())
+
+    @classmethod
+    def _read_int_keywords(cls, content_dict, config_content):
+        for key in cls._int_keywords:
+            if config_content.hasKey(key):
+                value = config_content.getValue(key)
+                if value > 0:
+                    # less than 0 in the config is equivalent to setting None
+                    # (backwards compatability)
+                    content_dict[key.lower()] = value
+
+    @classmethod
+    def from_config_file(cls, config_file: str, name: Optional[str] = None):
+        if name is None:
+            name = os.path.basename(config_file)
+
+        try:
+            config_content = cls._parse_config_file(config_file)
+        except IOError as err:
+            raise ConfigValidationError(
+                f"Could not open job config file {config_file}"
+            ) from err
+
+        content_dict = {}
+
+        cls._read_str_keywords(content_dict, config_content)
+        cls._read_int_keywords(content_dict, config_content)
+
+        content_dict["executable"] = config_content.getValue("EXECUTABLE")
+        if config_content.hasKey("ARGLIST"):
+            # We unescape backslash here to keep backwards compatability ie. If
+            # the arglist contains a '\n' we interpret it as a newline.
+            content_dict["arglist"] = [
+                s.encode("utf-8", "backslashreplace").decode("unicode_escape")
+                for s in config_content["ARGLIST"][-1]
+            ]
+
+        arg_types_dict = defaultdict(lambda: ContentTypeEnum.CONFIG_STRING)
+        if "max_arg" in content_dict and content_dict["max_arg"] > 0:
+            arg_types_dict[content_dict["max_arg"] - 1] = ContentTypeEnum.CONFIG_STRING
+        for arg in config_content["ARG_TYPE"]:
+            arg_types_dict[arg[0]] = ContentTypeEnum(type_from_kw(arg[1]))
+        if arg_types_dict:
+            content_dict["arg_types"] = [
+                arg_types_dict[j]
+                for j in range(max(i for i in arg_types_dict.keys()) + 1)
```

Comment:
i have so many questions. again i'm in a rush to leave, and it's late, so maybe i'm doing the dumbs, but why not just use something like `len(arg_typed_dict)`?

Reply:
Yea, that is not the same :(

Reply:
could you explain a bit? do / can we not expect every argument (index) to be present in the dict?

Reply:
The reason is that the file might not contain type information for all indecies.